packages/access-client/src/agent-use-cases.js

```diff
@@ -201,12 +239,28 @@ export async function authorizeAndWait(access, email, opts) {
       { can: 'upload/*' },
     ]
   )
-  const sessionDelegations = [...(await expectAuthorization())]
+  const sessionDelegations = [...(await expectAuthorization(access, opts))]
   if (!opts?.dontAddProofs) {
     await Promise.all(sessionDelegations.map(async (d) => access.addProof(d)))
   }
 }
 
+/**
+ * Request authorization of a session allowing this agent to issue UCANs
+ * signed by the passed email address.
+ *
+ * @param {AccessAgent} accessAgent
+ * @param {`${string}@${string}`} email
+ * @param {object} [opts]
+ * @param {AbortSignal} [opts.signal]
+ * @param {Iterable<{ can: Ucanto.Ability }>} [opts.capabilities]
+ * @param {AuthorizationWaiter} [opts.expectAuthorization] - function that will resolve once account has confirmed the authorization request
+ */
+export async function authorizeWaitAndClaim(accessAgent, email, opts) {
+  await authorizeAndWait(accessAgent, email, opts)
+  await claimAccess(accessAgent, accessAgent.issuer.did(), { addProofs: true })
```

Comment:
can we make the `addProofs` come from `opts`?
or call this `authorizeWaitAndClaimAndAdd` (claiming and adding are two separable operations)

Reply:
oo I like making it configurable yep